apps/web/src/components/user-tools/share/share-modal.tsx

```diff
@@ -46,18 +46,29 @@ export function ShareModal({
   const { strings, lang } = useInstanceData()
   const { entityId } = useEntityData()
   const pathOrId = path ?? entityId
+  const [isClipboardSupported, setIsClipboardSupported] = useState(false)
+
+  useEffect(() => {
+    setIsClipboardSupported(navigator.clipboard !== undefined)
+  }, [])
 
   if (!isOpen || !pathOrId) return null
 
-  function copyToClipboard(event: MouseEvent, text?: string) {
-    const target = event.target as HTMLAnchorElement
-    shareInputRef.current!.select()
-    document.execCommand('copy')
-    target.focus()
-    showToastNotice(
-      '👌 ' + (text ? text : strings.share.copySuccess),
-      'success'
-    )
+  async function copyToClipboard(event: MouseEvent, text?: string) {
+    try {
+      if (shareInputRef.current) {
+        await navigator.clipboard.writeText(shareInputRef.current.value)
```

Comment:
```suggestion
        await navigator.clipboard.writeText(shareUrl)
```

I think `shareInputRef` is generally no longer needed and can be removed.